sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -0,0 +1,410 @@
+            (_nib_get_if(tmp_node) == iface)) {
+            /* exact match */
+            DEBUG("  %p is an exact match\n", (void *)tmp);
+            tmp_node->mode |= _DRL;
+            return tmp;
+        }
+        if ((def_router == NULL) && (tmp_node == NULL)) {
+            def_router = tmp;
+        }
+    }
+    if (def_router != NULL) {
+        DEBUG("  using %p\n", (void *)def_router);
+        def_router->next_hop = _nib_alloc(router_addr, iface);
+
+        if (def_router->next_hop == NULL) {
+            return NULL;
+        }
+        _override_node(router_addr, iface, def_router->next_hop);
+        def_router->next_hop->mode |= _DRL;
+    }
+    return def_router;
+}
+
+void _nib_drl_remove(_nib_dr_t *nib_dr)
+{
+    if (nib_dr->next_hop != NULL) {
+        nib_dr->next_hop->mode &= ~(_DRL);
+        _nib_clear(nib_dr->next_hop);
+        memset(nib_dr, 0, sizeof(_nib_dr_t));
+    }
+    if (nib_dr == _prime_def_router) {
+        _prime_def_router = NULL;
+    }
+}
+
+_nib_dr_t *_nib_drl_iter(const _nib_dr_t *last)
+{
+    bool found = false;
+
+    for (unsigned i = 0; i < GNRC_IPV6_NIB_DEFAULT_ROUTER_NUMOF; i++) {
+        _nib_dr_t *def_router = &_def_routers[i];
+        _nib_t *node = def_router->next_hop;
+
+        if (node != NULL) {
+            /* It is linked to the default router list so it *should* be set */
+            assert(node->mode & _DRL);
+            if ((last == NULL)) {
+                return def_router;
+            }
+            else if (!found && (last == def_router)) {
+                found = true;
+            }
+            else if (found) {
+                return def_router;
+            }
+        }
+    }
+    return NULL;
+}
+
+_nib_dr_t *_nib_drl_get(const ipv6_addr_t *router_addr, unsigned iface)
+{
+    for (unsigned i = 0; i < GNRC_IPV6_NIB_DEFAULT_ROUTER_NUMOF; i++) {
+        _nib_dr_t *def_router = &_def_routers[i];
+        _nib_t *node = def_router->next_hop;
+
+        if ((node != NULL) &&
+            (ipv6_addr_equal(router_addr, &node->ipv6)) &&
+            (_nib_get_if(node) == iface)) {
+            /* It is linked to the default router list so it *should* be set */
+            assert(node->mode & _DRL);
+            return def_router;
+        }
+    }
+    return NULL;
+}
+
+_nib_dr_t *_nib_drl_get_dr(void)
+{
+    if ((_prime_def_router == NULL) ||
+        (_node_unreachable(_prime_def_router->next_hop))) {
+        _prime_def_router = NULL;   /* in case the prime router was unreachable */
+        _nib_dr_t *last = _last_def_router;
+        _nib_dr_t *res = _nib_drl_iter(last);
+
+        if (res == NULL) {
+            res = _nib_drl_iter(res);   /* try to wrap around */
+            if (res == NULL) {          /* we really don't have a default router */
+                return NULL;
+            }
+        }
+        /* search a reachable router */
+        while (_node_unreachable(res->next_hop)) {
```

Comment:
After an off-github discussion with martine, I realized that this function must return an unreachable default router in case there is no reachable router. I am fine with the current solution (: